diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 54e3459..de1bd60 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -1162,21 +1162,39 @@ install(FILES ${IFCPARSE_H_FILES}
 )
 
 install(TARGETS IfcParse
+    EXPORT IfcOpenShellTargets
 	ARCHIVE DESTINATION ${LIBDIR}
 	LIBRARY DESTINATION ${LIBDIR}
 	RUNTIME DESTINATION ${BINDIR}
 )
 
 if(BUILD_IFCGEOM)
-    install(FILES ${IFCGEOM_H_FILES}
-        DESTINATION ${INCLUDEDIR}/ifcgeom
+    file(GLOB IFCGEOM_HEADER_FILES ../src/ifcgeom/*.h)
+
+    install(FILES ${IFCGEOM_HEADER_FILES}
+       DESTINATION ${INCLUDEDIR}/ifcgeom
     )
 
-    install(FILES ${SCHEMA_AGNOSTIC_H_FILES}
-        DESTINATION ${INCLUDEDIR}/ifcgeom_schema_agnostic
+    file(GLOB IFCGEOM_SERIALIZATION_HEADER_FILES ../src/ifcgeom/Serialization/*h)
+
+    install(FILES ${IFCGEOM_SERIALIZATION_HEADER_FILES}
+        DESTINATION ${INCLUDEDIR}/ifcgeom/Serialization
+    )
+
+    file(GLOB IFCGEOM_OPENCASCADE_HEADER_FILES ../src/ifcgeom/kernels/opencascade/*h)
+
+    install(FILES ${IFCGEOM_OPENCASCADE_HEADER_FILES}
+        DESTINATION ${INCLUDEDIR}/ifcgeom/kernels/opencascade
+    )
+
+    file(GLOB IFCGEOM_MAPPING_HEADER_FILES ../src/ifcgeom/mapping/*h)
+
+    install(FILES ${IFCGEOM_MAPPING_HEADER_FILES}
+        DESTINATION ${INCLUDEDIR}/ifcgeom/mapping
     )
 
     install(TARGETS ${IFCGEOM_SCHEMA_LIBRARIES} ${kernel_libraries} IfcGeom
+        EXPORT IfcOpenShellTargets
         ARCHIVE DESTINATION ${LIBDIR}
         LIBRARY DESTINATION ${LIBDIR}
         RUNTIME DESTINATION ${BINDIR}
@@ -1184,21 +1202,41 @@ if(BUILD_IFCGEOM)
 endif(BUILD_IFCGEOM)
 
 if(BUILD_CONVERT)
-    install(TARGETS Serializers ${SERIALIZER_SCHEMA_LIBRARIES}
-        ARCHIVE DESTINATION ${LIBDIR}
-        LIBRARY DESTINATION ${LIBDIR}
-        RUNTIME DESTINATION ${BINDIR}
-    )
-
-    install(FILES ${SERIALIZERS_FILES}
+    install(FILES ${SERIALIZERS_H_FILES}
         DESTINATION ${INCLUDEDIR}/serializers/
     )
 
-    install(FILES ${SERIALIZERS_S_FILES}
+    install(FILES ${SERIALIZERS_S_H_FILES}
         DESTINATION ${INCLUDEDIR}/serializers/schema_dependent
     )
+
+    install(TARGETS Serializers ${SERIALIZER_SCHEMA_LIBRARIES} geometry_serializer ${geometry_serializer_libraries} 
+        EXPORT IfcOpenShellTargets
+        ARCHIVE DESTINATION ${LIBDIR}
+        LIBRARY DESTINATION ${LIBDIR}
+        RUNTIME DESTINATION ${BINDIR}
+    )
 endif(BUILD_CONVERT)
 
+install(EXPORT IfcOpenShellTargets
+    FILE IfcOpenShellTargets.cmake
+    NAMESPACE IfcOpenShell::
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake
+)
+
+include(CMakePackageConfigHelpers)
+
+configure_package_config_file(${CMAKE_CURRENT_SOURCE_DIR}/Config.cmake.in
+    "${CMAKE_CURRENT_BINARY_DIR}/IfcOpenShellConfig.cmake"
+    INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake
+)
+
+install(FILES
+    "${CMAKE_CURRENT_BINARY_DIR}/IfcOpenShellConfig.cmake"
+    #"${CMAKE_CURRENT_BINARY_DIR}/IfcOpenShellConfigVersion.cmake"
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake
+)
+
 # Cmake uninstall target
 if(NOT TARGET uninstall)
     configure_file(
